File: src/Emergence.Data.Shared/Search/Models/PlantInfo.cs
using System;
using System.Collections.Generic;

namespace Emergence.Data.Shared.Search.Models
{
    public class PlantInfo
    {
        public int Id { get; set; }
        public string ScientificName { get; set; }
        public string CommonName { get; set; }
        public bool? Preferred { get; set; }
        public short? MinimumBloomTime { get; set; }
        public short? MaximumBloomTime { get; set; }
        public IEnumerable<Month> BloomTimes { get; set; }
        public double? MinimumHeight { get; set; }
        public double? MaximumHeight { get; set; }
        public DistanceUnit? HeightUnit { get; set; }
        public double? MinimumSpread { get; set; }
        public double? MaximumSpread { get; set; }
        public DistanceUnit? SpreadUnit { get; set; }
        public WaterType? MinimumWater { get; set; }
        public WaterType? MaximumWater { get; set; }
        public IEnumerable<WaterType> WaterTypes { get; set; }
        public LightType? MinimumLight { get; set; }
        public LightType? MaximumLight { get; set; }
        public IEnumerable<LightType> LightTypes { get; set; }
        public IEnumerable<StratificationStage> StratificationStages { get; set; }
        public Zone MinimumZone { get; set; }
        public Zone MaximumZone { get; set; }
        public IEnumerable<Zone> Zones { get; set; }
        public IEnumerable<Shared.Models.WildlifeEffect> WildlifeEffects { get; set; }
        public string Notes { get; set; }
        public Visibility Visibility { get; set; }
        public string CreatedBy { get; set; }
        public string ModifiedBy { get; set; }
        public DateTime DateCreated { get; set; }
        public DateTime? DateModified { get; set; }
        public Lifeform Lifeform { get; set; }
        public Origin Origin { get; set; }
        public Taxon Taxon { get; set; }
        public User User { get; set; }
        public IEnumerable<PlantLocation> PlantLocations { get; set; }
        public IEnumerable<Synonym> Synonyms { get; set; }
    }
}

File: src/Emergence.Web/ClientApp/src/app/shared/models/plant-info.ts
import { ConservationStatus, Effect, LightType, LocationStatus, Month, SoilType, StratificationType, Unit, Visibility, WaterType, Wildlife } from "./enums";
import { Lifeform } from "./lifeform";
import { GeoLocation } from "./location";
import { Origin } from "./origin";
import { Photo } from "./photo";
import { Taxon } from "./taxon";
import { User } from "./user";

export class PlantInfo {
    plantInfoId: number;
    lifeformId: number;
    scientificName: string;
    commonName: string;
    bloomTime: BloomTime;
    height: Height;
    spread: Spread;
    requirements : Requirements;
    visibility: Visibility;
    name: string;
    notes: string;
    wildlifeEffects: WildlifeEffect[];
    userId: number;
    createdBy: string;
    dateCreated: Date;
    modifiedBy: string;
    dateModified: Date;
    origin: Origin;
    lifeform: Lifeform;
    taxon: Taxon;
    photos: Photo[];
    user: User;
    soilTypes: SoilType[];
    locations: PlantLocation[];
}
  
export class BloomTime {
  minimumBloomTime: Month;
  maximumBloomTime: Month;
}

export class Height {
  minimumHeight: number;
  maximumHeight: number;
  unit: Unit;
}

export class Spread {
  minimumSpread: number;
  maximumSpread: number;
  unit: Unit;
}

export class Requirements {
  waterRequirements: WaterRequirements;
  lightRequirements: LightRequirements;
  zoneRequirements: ZoneRequirements;
  stratificationStages: StratificationStage[];
}

export class WaterRequirements {
  minimumWater: WaterType;
  maximumWater: WaterType;
}

export class LightRequirements {
  minimumLight: LightType;
  maximumLight: LightType;
}

export class ZoneRequirements {
  minimumZone: Zone;
  maximumZone: Zone;
}

export class Zone {
  id: number;
  name: string;
}

export class StratificationStage {
  step: number;
  dayLength: number;
  stratificationType: StratificationType;
}

export class WildlifeEffect {
  wildlife: Wildlife;
  effect: Effect;
}

export class PlantLocation {
  location: GeoLocation;
  status: LocationStatus;
  conservationStatus: ConservationStatus;
  createdBy: string;
  dateCreated: Date;
  modifiedBy: string;
  dateModified: Date;
}